src/models/room.js

```diff
@@ -263,6 +263,33 @@ Room.prototype.getRecommendedVersion = async function() {
         }
     }
 
+    let result = this._checkVersionAgainstCapability(versionCap);
+    if (result.urgent && result.needsUpgrade) {
+        // Something doesn't feel right: we shouldn't need to update
+        // because the version we're on should be in the protocol's
+        // namespace. This usually means that the server was updated
+        // before the client was, making us think the newest possible
+        // room version is not stable. As a solution, we'll refresh
+        // the capability we're using to determine this.
+        logger.warn(
+            "Refreshing room version capability because the server looks " +
+            "to be supporting a newer room version we don't know about.",
+        );
+
+        const caps = await this._client.getCapabilities(true);
+        if (!caps) {
```

Comment:
getCapabilities seems to always return an object, so this branch would never run? Should it be `if (caps["m.room_versions"]) ...` rather?

Reply:
Does it actually? That's embarassing. Should be checking the room version capability, yes.